File: dot-net/src/ISP.Animal/ISP.Animal/Model/Serpiente.cs
using System;

namespace ISP.Animal.Model
{
    public class Serpiente : Animal
    {
        private bool _alimentada;

        public override void Alimentar()
        {
            _alimentada = true;
        }

        public override void Acariciar()
        {
            // ignore - I'm not grooming a freaking rattlesnake
            throw new NotImplementedException("No voy a acariciar a una serpiente");
        }

        public override string GetEstado()
        {
            var estado = "quiero una rata";

            if (_alimentada)
				estado = "no " + estado;

            return estado;
        }
    }
}

File: java/isp-animal/src/Serpiente.java

public class Serpiente extends Animal {

    private boolean alimentada = false;

    @Override
	public void alimentar() {
    	alimentada = true;
	}

	@Override
	public void acariciar() {
        // ignore - No voy a acariciar a una serpiente
	}

	@Override
	public String getEstado() {
        String estado = "quiero una rata";

        if (this.alimentada)
			estado = "no " + estado;

        return estado;
	}
}